### PR TITLE
Fixes and cleans up Pump-Up code.

### LINKED ISSUE #78287: Pump-up drug doesn't add the baton resistance trait
Local testing. 

## Reproduction:
1. Eat or inject 20u Pump-Up
2. Telescopic baton yourself

You get knocked down. VV yourself and notice you do not have the `baton_resistance` trait. 

Ephedrine and Stimulants still work fine, give the trait and prevent baton knockdown. 

Relevant code:
https://github.com/tgstation/tgstation/blob/46dde5f041192e37f02e66d82fc287f012eaf4d6/code/modules/reagents/chemistry/reagents/drug_reagents.dm#L317-L337

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Fixes the var names and checks on pumpup. Ensures we're clearing any moodlets gained from the drug if we are an assistant.

Removes some duplicate code that was breaking pumpup.

## Why It's Good For The Game

fixes https://github.com/tgstation/tgstation/issues/78287

Foix

fuax

betr coud

## Changelog
:cl:
fix: Makes sure pump-up properly grants the baton resistance trait.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 1]
```diff
@@ -367,12 +370,6 @@
 	name = "Maintenance Drugs"
 	chemical_flags = NONE
 
-/datum/reagent/drug/pumpup/on_mob_metabolize(mob/living/carbon/L)
-	var/obj/item/organ/internal/liver/liver = L.get_organ_slot(ORGAN_SLOT_LIVER)
-	if(HAS_TRAIT(liver, TRAIT_MAINTENANCE_METABOLISM))
-		L.add_mood_event("maintenance_fun", /datum/mood_event/maintenance_high)
-		metabolization_rate *= 0.8
-
```
Comment (Fikou) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:375]: i think this is meant to be for the maintenance drugs, they just forgot to change the typepath

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 2]
```diff
@@ -324,17 +324,20 @@
...
 		metabolization_rate *= 0.8
 
-/datum/reagent/drug/pumpup/on_mob_end_metabolize(mob/living/L)
-	REMOVE_TRAIT(L, TRAIT_BATON_RESISTANCE, type)
-	..()
+/datum/reagent/drug/pumpup/on_mob_end_metabolize(mob/living/affected_mob)
+	REMOVE_TRAIT(affected_mob, TRAIT_BATON_RESISTANCE, type)
+	var/obj/item/organ/internal/liver/liver = affected_mob.get_organ_slot(ORGAN_SLOT_LIVER)
+	if(liver && HAS_TRAIT(liver, TRAIT_MAINTENANCE_METABOLISM))
+		affected_mob.clear_mood_event("maintenance_fun")
```
Comment (Fikou) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:339]: why are you clearing it? it automatically deletes 2 minutes after its created

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 3]
```diff
@@ -324,17 +324,20 @@
...
+	. = ..()
```
Comment (Fikou) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:340]:
```suggestion
	return ..()
```

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 1 above)
Comment (carlarctg) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:375]: ye

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 2 above)
Comment (necromanceranne) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:339]: I'm just bloody silly is what

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:375]: Yeah that checks out.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**